pkg/controllermanager/controller/seed/seed_backup_bucket_check_control.go

```diff
@@ -57,41 +58,46 @@ func (c *Controller) backupBucketUpdate(oldObj, newObj interface{}) {
 		return
 	}
 
-	if !apiequality.Semantic.DeepEqual(oldBackupBucket.Status, newBackupBucket.Status) || !apiequality.Semantic.DeepEqual(oldBackupBucket.Spec, newBackupBucket.Spec) {
+	if lastErrorChanged(oldBackupBucket.Status.LastError, newBackupBucket.Status.LastError) {
 		c.backupBucketEnqueue(newBackupBucket)
 	}
 }
 
-// NewBackupBucketReconciler returns a new default control to checks backup buckets of related seeds.
-func NewBackupBucketReconciler(gardenClient client.Client) *backupBucketReconciler {
-	return &backupBucketReconciler{
+// NewBackupBucketsCheckReconciler creates a new reconciler that maintains the BackupBucketsReady condition of Seeds
+// according to the observed status of BackupBuckets.
+func NewBackupBucketsCheckReconciler(gardenClient client.Client, config config.SeedBackupBucketsCheckControllerConfiguration, clock clock.Clock) *backupBucketsCheckReconciler {
+	return &backupBucketsCheckReconciler{
 		gardenClient: gardenClient,
+		config:       config,
+		clock:        clock,
 	}
 }
 
-type backupBucketReconciler struct {
+type backupBucketsCheckReconciler struct {
 	gardenClient client.Client
+	config       config.SeedBackupBucketsCheckControllerConfiguration
+	clock        clock.Clock
 }
 
 type backupBucketInfo struct {
 	name     string
 	errorMsg string
 }
 
-func (b *backupBucketInfo) String() string {
+func (b *backupBucketInfo) string() string {
```

Comment:
I guess the idea initially was for the backupBucketInfo to implement the `Stringer` interface, but it wasn't used as such anywhere, and instead the `.String()` function was called directly here: https://github.com/gardener/gardener/blob/b49fb4e45c6de81bfcefbe0cbd0612577518d9e9/pkg/controllermanager/controller/seed/seed_backup_bucket_reconcile.go#L122

Anyway, I will fix it so that the implementation is done correctly and `Stringer` is used.